etc/config.xml

```diff
@@ -407,7 +407,7 @@
         <days_to_expire>1</days_to_expire>
         <payment_action>order</payment_action>
         <is_gateway>1</is_gateway>
-        <can_use_checkout>0</can_use_checkout>
+        <can_use_checkout>1</can_use_checkout>
```

Comment:
Did you check the response of Magento's internal payment-methods call? Pay-by-link shouldn't be in the available payment methods list for headless payments.